framework/include/materials/MaterialPropertyInterface.h

```diff
@@ -768,7 +768,11 @@ MaterialPropertyInterface::getPossiblyConstantGenericMaterialPropertyByName(
 {
   // Check if it's just a constant
   if (const auto * default_property = defaultGenericMaterialProperty<T, is_ad>(prop_name))
+  {
+    // This status is used to decide which object requested material properties
+    _get_material_property_called = true;
     return *default_property;
+  }
```

Comment:
```suggestion
  {
    _get_material_property_called = true;
    return *default_property;
  }
```
no longer done like that though I could go back. The problem is IPHDG stuff that calls getMatProp indirectly